overlord/devicestate/devicectx.go

```diff
@@ -30,6 +32,23 @@ func DeviceCtx(st *state.State, task *state.Task, providedDeviceCtx snapstate.De
 	if providedDeviceCtx != nil {
 		return providedDeviceCtx, nil
 	}
+
+	// see if we have a remodel in progress
```

Comment:
I haven't seen all the remodel branches yet so just guessing, but perhaps a `RemodelnProgress` helper would be useful (move the code into a helper). Of course if you see potential for it to be resued.